airbyte-integrations/connectors/source-google-sheets/source_google_sheets/source.py

```diff
@@ -206,7 +206,22 @@ def read(
             else:
                 logger.info(f"Skipping syncing sheet {sheet}: {reason}")
 
-        logger.info(f"Finished syncing spreadsheet {spreadsheet_id}")
+    def read(
+        self,
+        logger: AirbyteLogger,
+        config: json,
+        catalog: ConfiguredAirbyteCatalog,
+        state: Union[List[AirbyteStateMessage], MutableMapping[str, Any]] = None,
+    ) -> Generator[AirbyteMessage, None, None]:
+        try:
+            yield from self._read(logger, config, catalog, state)
+        except errors.HttpError as e:
+            if e.status_code == 429:
```

Comment:
Will records that were read before reaching the rate limits be successfully synced?

Reply:
Yes, I've tested it on our account. The current behavior is as follows, the records are read in the same way as in the previous version, when rate limits appear, backoff works. If rate limits still exist and backoff does not help, reading stops with log message. So instead of error we will stop sync and  show logs about rate limits.